File: 1.1/ruby/main.rb
#!/usr/bin/env ruby

lines = File.readlines('../../1.input')

def get_calibration_value(line)
  "#{line[/\d/]}#{line.reverse[/\d/]}".to_i
end

sum = lines.map { |line| get_calibration_value(line) }.reduce(:+)

puts sum

File: 1.1/python/main.py
#!/usr/bin/env python3

with open('../../1.input', 'r') as f:
    lines = f.read().splitlines()


def get_calibration_value(line):
    for char in line:
        if char.isdigit():
            first_digit = char
            break

    for char in reversed(line):
        if char.isdigit():
            last_digit = char
            break

    return int(first_digit + last_digit)


sum = sum([get_calibration_value(line) for line in lines])

print(sum)
